src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java

```diff
@@ -555,14 +546,23 @@
         return ResponseEntity.status(resource.getStatusCode()).body(resource);
     }
 
-    private String getUserEmail(String email) {
+    public String getUserEmailFromHeader(boolean isEmailRequired) {
```

Comment:
Please move this method to util class. Var isEmailRequired is not required now onwards since email header will be compulsory.  Remove log.warn () and replace else if with else directly.

Reply:
Addressed here: https://github.com/hmcts/rd-user-profile-api/pull/719/commits/f74f029a4f4f373ddecf6c61b907005b4d5cb4b9